src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -531,21 +531,15 @@ public ProgrammingExercise getExercise(SolutionProgrammingExerciseParticipation
      * 
      * @param id of the programming exercise.
      * @return
-     * @throws NoSuchElementException the programming exercise could not be found.
-     * @throws IllegalAccessException the retriever does not have the permissions to fetch information related to the programming exercise.
+     * @throws EntityNotFoundException the programming exercise could not be found.
      */
-    public ProgrammingExercise findById(Long id) throws NoSuchElementException, IllegalAccessException {
+    public ProgrammingExercise findById(Long id) throws EntityNotFoundException {
```

Comment:
please rename id to programmingExercseId

Reply:
Done!